LedgerSMB/Database/Loadorder.pm

```diff
@@ -0,0 +1,176 @@
+=head1 NAME
+
+LedgerSMB::Database::Loadorder - LOADORDER parsing
+
+=cut
+
+package LedgerSMB::Database::Loadorder;
+use strict;
+use warnings;
+
+use LedgerSMB::Database::Change;
+use Cwd;
+
+=head1 SYNOPSIS
+
+my $loadorder = LedgerSMB::Database::Loadorder->new('path/to/loadorder');
+my @scripts = $loadorder->scripts;
+
+# to make an index of scripts and their sha hashes for db revision checking:
+
+$loadorder->makeindex()
+
+But see the notes about locking below
+
+=head1 METHODS
+
+=head2 constructor
+
+LedgerSMB::Database::Loadorder->new($path);
+
+=cut
+
+sub new {
+    my ($package, $path) = @_;
+    bless {_path => $path }, $package;
+}
+
+=head2 scripts
+
+Returns a list of LedgerSMB::Database::Change objects
+
+=cut
+
+my $reload_subsequent;
+sub scripts {
+    my ($self) = @_;
+    return @{$self->{_scripts}} if $self->{_scripts};
+    my $loadorder;
+    local $!;
+    local $@;
+    open(LOAD, '<', $self->{_path}) or
```

Comment:
Do we put a restriction on file names? Like afaik only or utf8? In case of the latter, we want to open loadorder appropriately.